dev-lang/perl/perl-5.34.0.recipe

```diff
@@ -0,0 +1,127 @@
+tools, facilitating easy manipulation of text files.
+
+In addition to CGI, Perl is used for graphics programming, system \
+administration, network programming, finance, bioinformatics, and other \
+applications. Perl is nicknamed 'the Swiss Army chainsaw of scripting \
+languages' because of its flexibility and power, and possibly also because of \
+its perceived 'ugliness'."
+HOMEPAGE="https://www.perl.org/"
+COPYRIGHT="1993-2019 Larry Wall and others"
+LICENSE="GNU GPL v1
+	Artistic"
+REVISION="1"
+perlShortVersion="${portVersion%.*}"
+SOURCE_URI="http://www.cpan.org/src/perl-$portVersion.tar.gz"
+CHECKSUM_SHA256="551efc818b968b05216024fb0b727ef2ad4c100f8cb6b43fab615fa78ae5be9a"
+PATCHES="perl-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64 ?arm"
+
+PROVIDES="
+	# assume that any perl commands are backwards compatible to version 5.
+	perl = $portVersion compat >= 5
+	cmd:corelist = $portVersion compat >= 5
+	cmd:cpan = $portVersion compat >= 5
+	cmd:enc2xs = $portVersion compat >= 5
+	cmd:encguess = $portVersion compat >= 5
+	cmd:h2ph = $portVersion compat >= 5
+	cmd:h2xs = $portVersion compat >= 5
+	cmd:instmodsh = $portVersion compat >= 5
+	cmd:json_pp = $portVersion compat >= 5
+	cmd:libnetcfg = $portVersion compat >= 5
+	cmd:perl = $portVersion compat >= 5
+	cmd:perl$portVersion = $portVersion compat >= 5
+	cmd:perlbug = $portVersion compat >= 5
+	cmd:perldoc = $portVersion compat >= 5
+	cmd:perlivp = $portVersion compat >= 5
+	cmd:perlthanks = $portVersion compat >= 5
+	cmd:piconv = $portVersion compat >= 5
+	cmd:pl2pm = $portVersion compat >= 5
+	cmd:pod2html = $portVersion compat >= 5
+	cmd:pod2man = $portVersion compat >= 5
+	cmd:pod2text = $portVersion compat >= 5
+	cmd:pod2usage = $portVersion compat >= 5
+	cmd:podchecker = $portVersion compat >= 5
+	cmd:podselect = $portVersion compat >= 5
+	cmd:prove = $portVersion compat >= 5
+	cmd:ptar = $portVersion compat >= 5
+	cmd:ptardiff = $portVersion compat >= 5
+	cmd:ptargrep = $portVersion compat >= 5
+	cmd:shasum = $portVersion compat >= 5
+	cmd:splain = $portVersion compat >= 5
+	cmd:streamzip = $portVersion compat >= 5
+	cmd:xsubpp = $portVersion compat >= 5
+	cmd:zipdetails = $portVersion compat >= 5
+	# vendor_perl refers to the path to packaged perl modules, which includes
+	# the short perl version, so it is backwards compatible up this one.
+	vendor_perl = $perlShortVersion compat >= 5
+	lib:libperl = $portVersion compat >= $perlShortVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc
+	cmd:ld
+	cmd:make
+	cmd:sed
+	cmd:grep
+	"
+
+perlArchName="$(uname -m)-haiku"
+
+GLOBAL_WRITABLE_FILES="
+	"
+
+BUILD_PACKAGE_ACTIVATION_PHASE=INSTALL
+
+BUILD()
+{
+	./Configure -des -Dprefix=$prefix -Dsiteprefix=$prefix/non-packaged
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	chmod a+x $binDir/{perl,perlthanks}
+}
+
+TEST()
+{
+#x86_64
+#Failed 19 tests out of 2439, 99.22% okay.
```

Comment:
I do not updated this command, but there is still a bunch of failing tests but the list has evolved